### DISCUSSION TITLE
Correlation of constraints and rewards

### DISCUSSION THREAD

**Discussion by 99Abdel** [PKU-Alignment/safety-gymnasium#60] [Q&A]
I want to know where and in which form do you integrate the cost function of safety constraints into the overall training session?
If the constraint should not affect the rewards function how do you update the policy without being influenced by the constraints?
And where do they play a critical role?
do you have 2 separate policies? 
I would appreciate if you could guide me also in terms of code that you have written.
Thank you.

**Comment by muchvo** [comment]
This question is strongly related to algorithm rather than environments, because environments only provide information. If you want to figure out how algorithms handle observations from environments, you can check out [OmniSafe](https://github.com/PKU-Alignment/omnisafe). It is a comprehensive library for Safe RL.
